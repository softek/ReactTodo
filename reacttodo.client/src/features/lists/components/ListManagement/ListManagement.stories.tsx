import type { Meta, StoryObj } from '@storybook/react';
import {
  expect, fn, userEvent, waitFor, within,
} from '@storybook/test';
import { ListManagement } from './ListManagement';

const meta = {
  title: 'Features/ListManagement',
  component: ListManagement,
  parameters: {
    layout: 'centered',
  },
  argTypes: {
  },
  args: {
    onCreateList: fn(),
  },
} satisfies Meta<typeof ListManagement>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    lists: [
      { id: '1', name: 'List 1' },
      { id: '2', name: 'List 2' },
    ],
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);

    const inputBox = canvas.getByPlaceholderText('New list');
    await expect(inputBox).toBeInTheDocument();

    const createButton = canvas.getByRole('button', { name: /Create List/i });
    await expect(createButton).toBeInTheDocument();

    await userEvent.type(inputBox, 'New List Name');
    await userEvent.click(createButton);
    await waitFor(() => expect(args.onCreateList).toHaveBeenCalledWith('New List Name'));
    await expect(inputBox).toHaveValue('');
  },
};
