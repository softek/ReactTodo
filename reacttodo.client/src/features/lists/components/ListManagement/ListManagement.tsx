import React from 'react';
import { ListGroup, ListGroupItem } from 'reactstrap';
import { Link } from 'react-router-dom';
import { TodoList } from '../../../../data';
import { NewList } from '../NewList';

type ListManagementProps = {
  lists: TodoList[];
  onCreateList: (name: string) => void;
}

export function ListManagement({ lists, onCreateList }: ListManagementProps) {
  return (
    <div>
      <NewList onCreateList={(name) => onCreateList(name)} />
      <ListGroup>
        {lists.map((list) => (
          <ListGroupItem
            key={list.id}
            action
          >
            <Link to={`list/${list.id}`}>
              {list.name}
            </Link>
          </ListGroupItem>
        ))}
      </ListGroup>
    </div>
  );
}
