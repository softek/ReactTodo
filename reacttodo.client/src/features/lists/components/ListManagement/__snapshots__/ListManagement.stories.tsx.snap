// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features/ListManagement Default play-test 1`] = `
<div>
  <form>
    <div class="p-0 d-flex mb-4">
      <input placeholder="New list"
             name="name"
             type="text"
             class="form-control"
             aria-invalid="false"
             value
      >
      <button type="button"
              class="btn btn-success"
              aria-label="Create List"
      >
        Create List
      </button>
    </div>
  </form>
  <ul class="list-group">
    <li class="list-group-item-action list-group-item">
      <a href="/list/1">
        List 1
      </a>
    </li>
    <li class="list-group-item-action list-group-item">
      <a href="/list/2">
        List 2
      </a>
    </li>
  </ul>
</div>
`;
